File: javaspace/Solution14.java
import java.util.*;

// Codility - Distinct
public class Solution14 {
    public static void main(String[] args) {
        int[] A = {1,1,2,3,2,3};
        
        System.out.println(solution(A));
    }

    public static int solution(int[] A) {
        // write your code in Java SE 11
        HashSet<Integer> set = new HashSet<>();

        for (int i : A) {
            set.add(i);
        }

        return set.size();
    }
}

File: pythonspace/Solution14.py
# Codility - OddOccurrencesInArray

def solution(A):
    # write your code in Python 3.6
    if len(A) == 1:
        return A[0]

    A = sorted(A)
    print(A)
    for i in range(0, len(A), 2):
        if i+1 == len(A):
            return A[i]
        if A[i] != A[i+1]:
            return A[i]